Makefile

```diff
@@ -182,27 +182,53 @@ ifeq ($(USE_TLS),1)
   TLS_LDFLAGS+=-lssl -lcrypto
 endif
 
-ifeq ($(uname_S),FreeBSD)
+USE_THREADS?=1
+ifeq ($(USE_THREADS),1)
+  CFLAGS+=-DVALKEY_USE_THREADS
+endif
+
+PTHREAD_FLAGS :=
+
+ifeq ($(uname_S),Linux)
+  PTHREAD_FLAGS += -pthread
+else ifeq ($(uname_S),FreeBSD)
   REAL_LDFLAGS += -lm
+  ifeq ($(USE_THREADS),1)
+    PTHREAD_FLAGS += -pthread
+  endif
 else ifeq ($(uname_S),SunOS)
-  ifeq ($(shell $(CC) -V 2>&1 | grep -iq 'sun\|studio' && echo true),true)
+  ifeq ($(shell $(CC) -V 2>&1 grep -iq 'sun\|studio' && echo true),true)
```

Comment:
I wonder if there is some way to test this branch..
We have https://github.com/valkey-io/libvalkey/actions/runs/19833624895/job/56825506560
but that uses gcc

Reply:
You removed the pipe `|`? This looks like a typo to me. How can we add `grep` after `2>&1` without a pipe?